README.md

```diff
@@ -1,2 +1,276 @@
+
+1. Build projects directly from the command line
+
+      ```bash
+      cd ~/zxa
+      west build -p auto -b native_posix \
+        example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bash
+      ./build/example-application/zephyr/projects/zxa_example_project/native_posix/zephyr/zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### macOS
+
+1. Build projects directly from the command line
+
+      ```bash
+      cd ~/zxa
+      west build -p auto -b reel_board \
+        example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bash
+      ./build/example-application/zephyr/projects/zxa_example_project/native_posix/zephyr/zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### Windows
+
+1. Build projects directly from the command line
+
+      ```bat
+      cd %HOMEPATH%\zxa
+      west build -p auto -b native_posix \
+        example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bat
+      .\build\example-application\zephyr\projects\zxa_example_project\native_posix\zephyr\zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister \
+        -T example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister --coverage \
+        -T example-application\zephyr
+      ```
+
+
+# Module Contribution Guidelines
+
+## Design Principles
+
+The ZXA module is designed around the following principles:
+
+- Keep all Zephyr-specific content within the (required) `zephyr` folder.
+
+  - This enables adding the example on to an existing repository.
+
+- Follow the Zephyr repository structure, mechanisms, and rules
+  whenever possible.
+
+  - This minimizes the variance from what users and developers expect,
+    simplifies maintenance, and reuses existing solutions.
+
+- Extend the Zephyr repository structure to support an arbitrary number
```

Comment:
This can be added later or in a second, future, `example-complex-application` repo. We'll consider that, but let us start with something as simple as it gets.

Reply:
Now, please.